### PR TITLE
feat: adds classification_report in the ModelTester class

### PR DESCRIPTION
**Pull request by hasibzunair**
Show various metrics for classification problems. Per-class metrics are also available for multi-class classification use-cases.

### CODE REVIEW

File: decavision/model_testing/testing.py [hunk 1]
```diff
@@ -231,3 +231,21 @@ def evaluate(self, path):
         generator = self._load_dataset(path)
         results = self.model.evaluate(generator)
         print('Accuracy of', results[1] * 100, '%')
+    
+    def generate_classification_report(self, path):
+        """
+        Compute classification report resulting from predictions on a dataset of images.
```
Comment (yangobeil) [inline decavision/model_testing/testing.py]: Can you write the main quantities tha are displayed?

Comment (yangobeil) [reply decavision/model_testing/testing.py]: Also specify that the method prints the results.

**Comment by yangobeil** [comment]
Can you fix the issues mentionned in the bot's comment?

Can you test if it's necessary to import the efficientnets? I think it has to be done to load trained models. If it's the case you should add B7 to the imports and if not just remove the imports.

File: decavision/model_testing/testing.py (hunk 1 above)
Comment (hasibzunair) [reply decavision/model_testing/testing.py]: Done!

**Comment by hasibzunair** [comment]
Remove the imports, updated docstring and fixed bot comments.

**Comment by yangobeil** [comment]
> Remove the imports, updated docstring and fixed bot comments.

Thanks! So importing the efficientnets is not necessary?

Also there seems to be a few issues with PEP8 that are left.

**Comment by hasibzunair** [comment]
Yup. Not needed while loading models. Would have needed it if we saved it using `model.save_weights()`. I fixed the PEP8 issues, but can't check on my end if there's any left as I don't get any error messages.

**Comment by yangobeil** [comment]
> Yup. Not needed while loading models. Would have needed it if we saved it using `model.save_weights()`. I fixed the PEP8 issues, but can't check on my end if there's any left as I don't get any error messages.

Take a look at the comment on the PR made by pep8speaks. It analyses the code and reports PEP8 problems. It updates every time you make a push so you can make sure there are no issues left.

**Comment by hasibzunair** [comment]
Thanks, I did not know that. All done! Is there any tool to automate this? So that the whole script satisfies PEP8 style.

**Comment by yangobeil** [comment]
> Thanks, I did not know that. All done! Is there any tool to automate this? So that the whole script satisfies PEP8 style.

The bot that comments is automated. I decided to have it scan only the files modified in a PR but it could scan everything every time. There are tools to make the modifications automatically, via pre-commits. However I don't like to force people to do the modifications all the time so I prefer to just use the bot.

**Review by yangobeil — APPROVED** [review]
Good job!

**Comment by hasibzunair** [comment]
Thanks!

**Pull request merged**